lib/modules/manager/tekton/extract.ts

```diff
@@ -80,6 +85,80 @@ function getDeps(doc: TektonResource): PackageDependency[] {
   return deps;
 }
 
+function addPipelineAsCodeAnnotations(
+  annotations:
+    | {
+        [key: string]: string;
+      }
+    | undefined,
+  deps: PackageDependency[],
+): void {
+  if (is.nullOrUndefined(annotations)) {
+    return;
+  }
+
+  for (const annotationsKey in annotations) {
+    if (
+      !annotationsKey.toLowerCase().includes('pipelinesascode.tekton.dev/task')
+    ) {
+      continue;
+    }
+
+    const tasks = annotations[annotationsKey]
+      .replace('[', '')
+      .replace(']', '')
+      .split(',');
+    for (const task of tasks) {
+      const dep = getAnnotationDep(task.trim());
+      if (!dep) {
+        continue;
+      }
+      deps.push(dep);
+    }
+  }
+}
+
+const githubRelease = regEx(
+  /^(?<url>(?:(?:http|https):\/\/)?(?<path>(?:[^:/\s]+[:/])?(?<project>[^/\s]+\/[^/\s]+)))\/releases\/download\/(?<currentValue>.+)\/(?<subdir>[^?\s]*)$/,
+);
+
+const gitUrl = regEx(
+  /^(?<url>(?:(?:http|https):\/\/)?(?<path>(?:[^:/\s]+[:/])?(?<project>[^/\s]+\/[^/\s]+)))(?:\/raw)?\/(?<currentValue>.+?)\/(?<subdir>[^?\s]*)$/,
+);
+
+function getAnnotationDep(url: string): PackageDependency | null {
+  const dep: PackageDependency = {};
+  dep.depType = 'tekton-annotation';
+
+  if (githubRelease.test(url)) {
+    const match = githubRelease.exec(url);
```

Comment:
move exec outside of the condition and check match?.groups on condition to not execute regex twice